healthcheck/healthcheck.go

```diff
@@ -23,34 +25,41 @@ type HealthCheck struct {
 	Uptime                   time.Duration `json:"uptime"`
 	StartTime                time.Time     `json:"start_time"`
 	Checks                   []Check       `json:"checks"`
+	Started                  bool          `json:"-"`
+	Interval                 time.Duration `json:"-"`
 	Clients                  []*Client     `json:"-"`
 	CriticalErrorTimeout     time.Duration `json:"-"`
 	TimeOfFirstCriticalError time.Time     `json:"-"`
 	tickers                  []*ticker     `json:"-"`
 }
 
 // Create returns a new instantiated HealthCheck object. Caller to provide:
-// context and should utilise contextWithCancel
 // version information of the app,
 // criticalTimeout for how long to wait until an unhealthy dependent propagates its state to make this app unhealthy
 // interval in which to check health of dependencies
 // clients of type Client which contain a Checker function which is run to check the health of dependent apps
-func Create(ctx context.Context, version string, criticalTimeout, interval time.Duration, clients []*Client) HealthCheck {
-	tickers := newTickers(interval, clients)
+func Create(version string, criticalTimeout, interval time.Duration, clients []*Client) HealthCheck {
 
 	hc := HealthCheck{
+		Started:              false,
 		Clients:              clients,
 		Version:              version,
-		StartTime:            time.Now().UTC(),
 		CriticalErrorTimeout: criticalTimeout,
-		tickers:              tickers,
+		Interval:             interval,
 	}
-
-	hc.start(ctx)
-
 	return hc
 }
 
+// AddClient adds a provided client to the healthcheck
+func (hc *HealthCheck) AddClient(c *Client) {
+	log.Info("AddClient hit", log.Data{"client": c})
```

Comment:
this log line could be more descriptive if it stays, something like 'adding client to healthcheck' etc. I'm also not sure the merit of logging the client, because presumably it'll be a huge log of the entire client structure? is there a piece of information available that would be a clearer context?